### PR TITLE
DIV 3749 amend petition dynamic fee

### PR DESCRIPTION
**Pull request by mounikahmcts**
https://tools.hmcts.net/jira/browse/DIV-3749

Petitioner is submitting an amended petition and wants to pay the appropriate fee.

### CODE REVIEW

File: app/steps/help/need-help/index.js [hunk 1]
```diff
@@ -14,10 +16,16 @@ module.exports = class NeedHelpWithFees extends ValidationStep {
     };
   }
 
+  interceptor(ctx, session) {
+    ctx.feeToBePaid = parseBool(config.features.release520) && session.previousCaseId ? config.commonProps.amendFee.amount : config.commonProps.applicationFee.amount;
```
Comment (hosainnet) [inline app/steps/help/need-help/index.js]: Not sure why we need the flag here, is the fee value changing based on 5.2 release?

File: app/services/feesAndPaymentsService.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
 };
 
-module.exports = { get };
+const get = () => {
+  return getFee('petition-issue-fee');
+};
+
+module.exports = {
+  get,
+  getFee
```
[the remark this answers is not in the document]
Comment (mounikahmcts) [reply app/services/feesAndPaymentsService.js]: get is already used in other places. so do not want to disturb it. By default get will fetch issue application fee .

File: app/steps/help/need-help/index.js (hunk 1 above)
Comment (mounikahmcts) [reply app/steps/help/need-help/index.js]: This flag is used in displaying the amend fees in the content and it should be applicable only for 520 release.

File: app/services/feesAndPaymentsService.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (hosainnet) [reply app/services/feesAndPaymentsService.js]: Actually as far as I can tell, it's no logner used following the change on this line https://github.com/hmcts/div-petitioner-frontend/pull/419/files#diff-875652841cbd89dbeb763952bc590381L27

[the remark this answers is not in the document]
Comment (mounikahmcts) [reply app/services/feesAndPaymentsService.js]: True, I thought that it had references. Removed get

[the remark this answers is not in the document]
Comment (hosainnet) [reply app/services/feesAndPaymentsService.js]: 👍

File: app/middleware/updateApplicationFeeMiddleware.js [hunk 3]
```diff
@@ -21,18 +31,18 @@ redisClient.on('error', error => {
   logger.errorWithReq(null, 'redis_error', 'Failed to connect to Redis', error.message);
 });
 
-const getFeeFromFeesAndPayments = req => {
+const getFeeFromFeesAndPayments = feeName => {
```
Comment (hosainnet) [inline app/middleware/updateApplicationFeeMiddleware.js]: This function used to take req to be passed to logger, so actually now it needs 2 args. Req and feeName

File: app/middleware/updateApplicationFeeMiddleware.js [hunk 4]
```diff
@@ -44,7 +54,26 @@ const updateApplicationFeeMiddleware = (req, res, next) => {
         CONF.commonProps.applicationFee = JSON.parse(response);
         return true;
       }
-      return getFeeFromFeesAndPayments(req);
+      return getFeeFromFeesAndPayments('applicationFee');
```
Comment (hosainnet) [inline app/middleware/updateApplicationFeeMiddleware.js]: This should pass req as the first arg

File: app/middleware/updateApplicationFeeMiddleware.js [hunk 5]
```diff
@@ -44,7 +54,26 @@ const updateApplicationFeeMiddleware = (req, res, next) => {
...
+
+
+const updateAmendFeeMiddleware = (req, res, next) => {
+  redisClient.get('commonProps.amendFee.amount')
+    .then(response => {
+      if (response) {
+        CONF.commonProps.amendFee = JSON.parse(response);
+        return true;
+      }
+      return getFeeFromFeesAndPayments('amendFee');
```
Comment (hosainnet) [inline app/middleware/updateApplicationFeeMiddleware.js]: And here

File: app/middleware/updateApplicationFeeMiddleware.js (hunk 3 above)
Comment (mounikahmcts) [reply app/middleware/updateApplicationFeeMiddleware.js]: Added req object as well to the arguments and for logger

File: app/middleware/updateApplicationFeeMiddleware.js (hunk 4 above)
Comment (mounikahmcts) [reply app/middleware/updateApplicationFeeMiddleware.js]: Added req object as well to the arguments and for logger

**Comment by dtaylor7** [comment]
merged into https://github.com/hmcts/div-petitioner-frontend/pull/386

File: app/middleware/updateApplicationFeeMiddleware.js (hunk 5 above)
Comment (dtaylor7) [reply app/middleware/updateApplicationFeeMiddleware.js]: @hosainnet this has now been merged into https://github.com/hmcts/div-petitioner-frontend/pull/386 can you check these comments have been resolved on that ticket

File: app/middleware/updateApplicationFeeMiddleware.js (hunk 3 above)
Comment (dtaylor7) [reply app/middleware/updateApplicationFeeMiddleware.js]: @hosainnet this has now been merged into https://github.com/hmcts/div-petitioner-frontend/pull/386 can you check these comments have been resolved on that ticket

Comment (hosainnet) [reply app/middleware/updateApplicationFeeMiddleware.js]: Okay, I'll take a look. Why has it been merged there though? It's not a related feature

Comment (dtaylor7) [reply app/middleware/updateApplicationFeeMiddleware.js]: yes it is, all part of amend petition

File: app/middleware/updateApplicationFeeMiddleware.js (hunk 5 above)
Comment (hosainnet) [reply app/middleware/updateApplicationFeeMiddleware.js]: I see, it could have still been merged separately though, no? Just makes it easier to review etc

Comment (dtaylor7) [reply app/middleware/updateApplicationFeeMiddleware.js]: vivek asked to have them one pr to test

Comment (hosainnet) [reply app/middleware/updateApplicationFeeMiddleware.js]: Fair enough 👍

**Pull request closed without merging**